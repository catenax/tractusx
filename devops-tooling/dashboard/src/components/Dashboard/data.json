{
  "nodes": [
    {
      "id": 1,
      "name": "Broker"
    },
    {
      "id": 2,
      "name": "Connector 1"
    },
    {
      "id": 3,
      "name": "Connector 2"
    },
    {
      "id": 4,
      "name": "Connector 3"
    }
  ],
  "links": [
    {
      "source": 0,
      "target": 1
    },
    {
      "source": 0,
      "target": 2
    },
    {
      "source": 0,
      "target": 3
    }
  ]
}
