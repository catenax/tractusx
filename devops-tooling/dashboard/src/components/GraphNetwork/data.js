export let data =  {

    "nodes" :[
        {"id": "Mango", "group": 1},
        {"id": "Apple", "group": 1},
        {"id": "Orange", "group": 1},
        {"id": "Pinapple", "group": 1},
        {"id": "Strawberry", "group": 1},
        {"id": "Watermelon", "group": 1},
    ],

    "links": [
        {"source": "Mango", "target": "Apple", "value": 1},
        {"source": "Apple", "target": "Orange", "value": 8},
        {"source": "Orange", "target": "Pinapple", "value": 10},
        {"source": "Pinapple", "target": "Watermelon", "value": 6},
        {"source": "Pinapple", "target": "Strawberry", "value": 6},
    ]
}