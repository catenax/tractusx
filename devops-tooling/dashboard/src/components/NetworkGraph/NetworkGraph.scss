.container {
  width: 100%;
  height: 80vh;
  position: relative;

  svg {
    display: block;
    width: 100%;
    height: 100%;
  }
}

.male {
  fill: #000;
}

.female {
  fill: rgb(104, 10, 10);
}

.node text {
  font: 12px sans-serif;
}

div.tooltip {
  position: absolute;
  text-align: center;
  width: 110px;
  padding: 10px;
  font: 12px sans-serif;
  background: lightsteelblue;
  border: 0;
  border-radius: 8px;
  pointer-events: none;
}
