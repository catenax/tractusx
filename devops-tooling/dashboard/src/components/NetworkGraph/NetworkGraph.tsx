import * as d3 from 'd3';
import { Simulation, SimulationNodeDatum } from 'd3';
import React, { useState, useEffect, useRef } from 'react'
import Nodes from '../Nodes/Nodes';

export default function NetworkGraph(props) {
  const ref = useRef(null);
  const width = 640;
  const height = 480;
  let simulation: any;
  let svg: any;
  let nodes: any;
  let links: any;
  let labels: any;

  useEffect(() => {
    createForceLayout();
  })

  const createForceLayout = () => {
    svg = d3.select(ref.current).append('svg')
      .attr("class", "graph")
      .attr('viewBox', `${-width/2} ${-height/2} ${width} ${height}`)
      .attr('width', width)
      .attr('height', height);

    simulation = d3.forceSimulation(props.data.nodes)
      .force("link", d3.forceLink(props.data.links))
      .force("charge", d3.forceManyBody().strength(-5000)) // This adds repulsion between nodes. Play with the -400 for the repulsion strength 
      .force("x", d3.forceX())
      .force("y", d3.forceY());
    
    links = svg
      .append("g")
      .attr("stroke", "#999")
      .attr("stroke-opacity", 0.6)
      .selectAll("line")
      .data(props.data.links)
      .join("line")
      .attr("stroke-width", '2');
    
    nodes = new Nodes(svg, props.data.nodes);

    labels = svg.append("g")
      .attr("class", "labels")
      .selectAll("text")
      .data(props.data.nodes)
      .enter()
      .append("text")
      .attr('text-anchor', 'middle')
      .attr('dominant-baseline', 'central')
      .text((d: any) => d.name)
    
    simulation.on("tick", () => {
      positionForceElements();
    });
    nodes.items.call(drag(simulation));
    labels.call(drag(simulation));
  }

  function positionForceElements() {
    //update link positions
    links
      .attr("x1", (d: any) => d.source.x)
      .attr("y1", (d: any) => d.source.y)
      .attr("x2", (d: any) => d.target.x)
      .attr("y2", (d: any) => d.target.y);

  // update node positions
    nodes.items
      .attr("cx", (d: any) => d.x)
      .attr("cy", (d: any) => d.y);

    labels
      .attr("x", (d: any) => d.x)
      .attr("y", (d: any) => d.y);
  }

  function drag(simulation:Simulation<SimulationNodeDatum, undefined>) {    
    function dragstarted(event: CustomEvent) {
      if (!event.active) simulation.alphaTarget(0.3).restart();
      event.subject.fx = event.subject.x;
      event.subject.fy = event.subject.y;
    }
    
    function dragged(event: CustomEvent) {
      event.subject.fx = event.x;
      event.subject.fy = event.y;
    }
    
    function dragended(event: CustomEvent) {
      if (!event.active) simulation.alphaTarget(0);
      event.subject.fx = null;
      event.subject.fy = null;
    }
    
    return d3.drag()
      .on("start", dragstarted)
      .on("drag", dragged)
      .on("end", dragended);
  }

  return (
    <div className="graph" ref={ref}>
    </div>
  )
}

interface CustomEvent {
  subject: {
    fx: any; 
    fy: any;
    x:any;
    y:any;
  };
  x: any;
  y: any; 
  active:any;
}
