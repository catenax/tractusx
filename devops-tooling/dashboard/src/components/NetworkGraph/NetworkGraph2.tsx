import renderForceGraph from '../../d3/Force2';
import { useEffect } from 'react';


export default function NetworkGraph(props) {


  useEffect(() => {
    renderForceGraph(props.nodes, props.links,'network-graph');
  },[props.nodes,props.links])

  return ( <div id="network-graph" style={{height:'100vh',width:'100%',backgroundColor:'grey'}}> </div> );
}