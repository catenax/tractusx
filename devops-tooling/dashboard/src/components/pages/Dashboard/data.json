{
  "nodes": [
    {
      "id": 1,
      "name": "Broker"
    },
    {
      "id": 2,
      "name": "Connector 1"
    },
    {
      "id": 3,
      "name": "Connector 2"
    },
    {
      "id": 4,
      "name": "Connector 3"
    }
  ],
  "links": [
    {
      "id": 1,
      "source": 0,
      "target": 1,
      "status": "inactive"
    },
    {
      "id": 2,
      "source": 0,
      "target": 2
    },
    {
      "id": 3,
      "source": 0,
      "target": 3
    }
  ]
}
