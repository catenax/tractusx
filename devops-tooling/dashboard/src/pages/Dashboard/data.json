{
  "nodes": [
    {
      "id": 1,
      "name": "Dataspace Connector"
    },
    {
      "id": 2,
      "name": "Connector 1"
    },
    {
      "id": 3,
      "name": "Connector 2"
    },
    {
      "id": 4,
      "name": "Connector 3"
    }
  ],
  "links": [
    {
      "id": 1,
      "source": 1,
      "target": 2,
      "issued": "2021-12-02T07:11:55.825Z"
    },
    {
      "id": 2,
      "source": 1,
      "target": 3,
      "issued": "2021-11-25T10:11:55.825Z"
    },
    {
      "id": 3,
      "source": 1,
      "target": 4,
      "issued": "2021-12-10T10:11:55.825Z"
    },
    {
      "id": 4,
      "source": 2,
      "target": 3,
      "issued": "2021-12-02T07:11:55.825Z"
    },
    {
      "id": 5,
      "source": 4,
      "target": 3,
      "issued": "2021-12-02T07:11:55.825Z"
    },
    {
      "id": 5,
      "source": 2,
      "target": 4,
      "issued": "2021-12-02T07:11:55.825Z"
    }
  ]
}
