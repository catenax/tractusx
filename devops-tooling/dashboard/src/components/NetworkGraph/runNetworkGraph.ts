
import * as d3 from "d3";
//import "@fortawesome/fontawesome-free/css/all.min.css";
import "./NetworkGraph.scss";

export function runNetworkGraph(
  container: any,
  linksData: any,
  nodesData: any,
  nodeHoverTooltip: any
) {
  const links = linksData.map((d: any) => Object.assign({}, d));
  const nodes = nodesData.map((d: any) => Object.assign({}, d));

  console.log(links)
  console.log(nodes)

  const containerRect = container.getBoundingClientRect();
  const height = containerRect.height;
  const width = containerRect.width;

  const color = () => { return "#9D79A0"; };

  const icon = (d: any) => {
    return d.gender === "male" ? "male" : "female";
  }

  const getClass = (d: any) => {
    return d.gender === "male" ? 'male' : 'female';
  };

  const drag = (simulation: any) => {
    const dragstarted = (event: any, d: any) => {
      if (!event.active) simulation.alphaTarget(0.3).restart();
      d.fx = d.x;
      d.fy = d.y;
    };

    const dragged = (event: any, d: any) => {
      d.fx = event.x;
      d.fy = event.y;
    };

    const dragended = (event: any, d: any) => {
      if (!event.active) simulation.alphaTarget(0);
      d.fx = null;
      d.fy = null;
    };

    return d3
      .drag()
      .on("start", dragstarted)
      .on("drag", dragged)
      .on("end", dragended);
  };

  // Add the tooltip element to the graph
  const tooltip = document.querySelector("#graph-tooltip");
  if (!tooltip) {
    const tooltipDiv = document.createElement("div");
    tooltipDiv.classList.add('tooltip');
    tooltipDiv.style.opacity = "0";
    tooltipDiv.id = "graph-tooltip";
    document.body.appendChild(tooltipDiv);
  }
  const div = d3.select("#graph-tooltip");

  const addTooltip = (hoverTooltip: any, d: any, x: any, y: any) => {
    div
      .transition()
      .duration(200)
      .style("opacity", 0.9);
    div
      .html(hoverTooltip(d))
      .style("left", `${x}px`)
      .style("top", `${y - 28}px`);
  };

  const removeTooltip = () => {
    div
      .transition()
      .duration(200)
      .style("opacity", 0);
  };

  const simulation = d3
    .forceSimulation(nodes)
    .force("charge", d3.forceManyBody().strength(-150))
    .force("link", d3.forceLink(links))
    .force("x", d3.forceX())
    .force("y", d3.forceY());

  const svg = d3
    .select(container)
    .append("svg")
    .attr("viewBox", `${-width / 2} ${-height / 2} ${width} ${height}`)
/*     .call(d3.zoom().on("zoom", function (event) {
      svg.attr("transform", event.transform);
    })); */

  const link = svg
    .append("g")
    .attr("stroke", "#999")
    .attr("stroke-opacity", 0.6)
    .selectAll("line")
    .data(links)
    .join("line")
    .attr("stroke-width", '10');

  const node = svg
    .append("g")
    .attr("stroke", "#fff")
    .attr("stroke-width", 2)
    .selectAll("circle")
    .data(nodes)
    .join("circle")
    .attr("r", 30)
    .attr("fill", color)
    //.call(drag(simulation));

  const label = svg.append("g")
    .attr("class", "labels")
    .selectAll("text")
    .data(nodes)
    .enter()
    .append("text")
    .attr('text-anchor', 'middle')
    .attr('dominant-baseline', 'central')
    .attr("class", (d: any) => d.gender)
    .text((d: any) => d.name)
    //.call(drag(simulation));

  label.on("mouseover", (event, d) => {
    addTooltip(nodeHoverTooltip, d, event.pageX, event.pageY);
  })
    .on("mouseout", () => {
      removeTooltip();
    });

  simulation.on("tick", () => {
    //update link positions
    link
      .attr("x1", (d: any) => d.source.x)
      .attr("y1", (d: any) => d.source.y)
      .attr("x2", (d: any) => d.target.x)
      .attr("y2", (d: any) => d.target.y);

    // update node positions
    node
      .attr("cx", (d: any) => d.x)
      .attr("cy", (d: any) => d.y);

    // update label positions
    label
      .attr("x", (d: any) => { return d.x; })
      .attr("y", (d: any) => { return d.y; })
  });

  return {
    destroy: () => {
      simulation.stop();
    },
    nodes: () => {
      return svg.node();
    }
  };
}
