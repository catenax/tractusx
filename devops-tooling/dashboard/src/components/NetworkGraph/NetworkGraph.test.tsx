
import { render,screen } from "@testing-library/react";
import NetworkGraph from "./NetworkGraph";
import data from '../../pages/Dashboard/data.json';

const nodesData = data.nodes;
const linksData = data.links;
const parentSize = {width:100,height:100};

describe("NetworkGraph rendering", () => {

  test('it renders connectors', () => {
    render(<NetworkGraph nodes={nodesData} links={linksData} parentSize={parentSize} />);
    const connectorElement = screen.getByText('Connector 1');
    expect(connectorElement).toBeInTheDocument();
    const connectorElement2 = screen.getByText('Connector 2');
    expect(connectorElement2).toBeInTheDocument();
  });

  test('it matches snapshots',()=> {
    let graphRender =  render(<NetworkGraph nodes={nodesData} links={linksData} parentSize={parentSize} />);
    expect(graphRender).toMatchSnapshot();
  })

});

describe("Warning functionality", () => {

  test('it renders a warning icon and changes color of Node', () => {
    nodesData[1]['status']  = {type: 'warning', text: 'The connection has been interrupted.'};
    render(<NetworkGraph nodes={nodesData} links={linksData} parentSize={parentSize} />);
    const warningTooltip = screen.getByText('The connection has been interrupted.');
    const warningIcon = screen.getByText('\uf071');
    expect(warningTooltip).toBeInTheDocument();
    expect(warningIcon).toBeInTheDocument();
  });
});
