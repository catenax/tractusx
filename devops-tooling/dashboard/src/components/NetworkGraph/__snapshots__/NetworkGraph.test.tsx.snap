// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkGraph rendering it matches snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="network-graph"
        style="background-color: rgb(119, 119, 119); height: 100%;"
      >
        <svg
          height="100"
          style="max-width: 100%; height: auto; height: intrinsic;"
          viewBox="0,0,100,100"
          width="100"
        >
          <g
            stroke="#fff"
            stroke-linecap="round"
            stroke-opacity="0.6"
            stroke-width="2"
          >
            <line />
            <line />
            <line />
            <line />
            <line />
            <line />
          </g>
          <g
            class="nodes"
          >
            <g>
              <circle
                fill="#69b3a2"
                r="30"
                stroke="#fff"
                stroke-opacity="1"
                stroke-width="2"
              />
              <text
                dominant-baseline="central"
                fill="black"
                text-anchor="middle"
              >
                <tspan>
                  Broker 12 
                </tspan>
                 
              </text>
              <title>
                0
              </title>
            </g>
            <g>
              <circle
                fill="#69b3a2"
                r="30"
                stroke="#fff"
                stroke-opacity="1"
                stroke-width="2"
              />
              <text
                dominant-baseline="central"
                fill="black"
                text-anchor="middle"
              >
                <tspan>
                  Broker 
                </tspan>
                 
              </text>
              <title>
                1
              </title>
            </g>
            <g>
              <circle
                fill="#69b3a2"
                r="30"
                stroke="#fff"
                stroke-opacity="1"
                stroke-width="2"
              />
              <text
                dominant-baseline="central"
                fill="black"
                text-anchor="middle"
              >
                <tspan>
                  Connector 1 
                </tspan>
                 
              </text>
              <title>
                2
              </title>
            </g>
            <g>
              <circle
                fill="#69b3a2"
                r="30"
                stroke="#fff"
                stroke-opacity="1"
                stroke-width="2"
              />
              <text
                dominant-baseline="central"
                fill="black"
                text-anchor="middle"
              >
                <tspan>
                  Connector 2 
                </tspan>
                 
              </text>
              <title>
                3
              </title>
            </g>
            <g>
              <circle
                fill="#69b3a2"
                r="30"
                stroke="#fff"
                stroke-opacity="1"
                stroke-width="2"
              />
              <text
                dominant-baseline="central"
                fill="black"
                text-anchor="middle"
              >
                <tspan>
                  Connector 3 
                </tspan>
                 
              </text>
              <title>
                4
              </title>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="network-graph"
      style="background-color: rgb(119, 119, 119); height: 100%;"
    >
      <svg
        height="100"
        style="max-width: 100%; height: auto; height: intrinsic;"
        viewBox="0,0,100,100"
        width="100"
      >
        <g
          stroke="#fff"
          stroke-linecap="round"
          stroke-opacity="0.6"
          stroke-width="2"
        >
          <line />
          <line />
          <line />
          <line />
          <line />
          <line />
        </g>
        <g
          class="nodes"
        >
          <g>
            <circle
              fill="#69b3a2"
              r="30"
              stroke="#fff"
              stroke-opacity="1"
              stroke-width="2"
            />
            <text
              dominant-baseline="central"
              fill="black"
              text-anchor="middle"
            >
              <tspan>
                Broker 12 
              </tspan>
               
            </text>
            <title>
              0
            </title>
          </g>
          <g>
            <circle
              fill="#69b3a2"
              r="30"
              stroke="#fff"
              stroke-opacity="1"
              stroke-width="2"
            />
            <text
              dominant-baseline="central"
              fill="black"
              text-anchor="middle"
            >
              <tspan>
                Broker 
              </tspan>
               
            </text>
            <title>
              1
            </title>
          </g>
          <g>
            <circle
              fill="#69b3a2"
              r="30"
              stroke="#fff"
              stroke-opacity="1"
              stroke-width="2"
            />
            <text
              dominant-baseline="central"
              fill="black"
              text-anchor="middle"
            >
              <tspan>
                Connector 1 
              </tspan>
               
            </text>
            <title>
              2
            </title>
          </g>
          <g>
            <circle
              fill="#69b3a2"
              r="30"
              stroke="#fff"
              stroke-opacity="1"
              stroke-width="2"
            />
            <text
              dominant-baseline="central"
              fill="black"
              text-anchor="middle"
            >
              <tspan>
                Connector 2 
              </tspan>
               
            </text>
            <title>
              3
            </title>
          </g>
          <g>
            <circle
              fill="#69b3a2"
              r="30"
              stroke="#fff"
              stroke-opacity="1"
              stroke-width="2"
            />
            <text
              dominant-baseline="central"
              fill="black"
              text-anchor="middle"
            >
              <tspan>
                Connector 3 
              </tspan>
               
            </text>
            <title>
              4
            </title>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
